packages/api/src/car.js

```diff
@@ -137,12 +139,7 @@ export async function carPost (request, env, ctx) {
 
   const blob = await request.blob()
   const bytes = new Uint8Array(await blob.arrayBuffer())
-  const reader = await CarReader.fromBytes(bytes)
-
-  const chunkSize = await getBlocksSize(reader)
-  if (chunkSize === 0) {
-    throw new Error('empty CAR')
-  }
+  const stat = await carStat(bytes)
```

Comment:
CAR could be LARGE at this point. We could pass a clone of the request to stat and have it convert the request body into an AsyncIterable so that we can use CarBlockIterator.fromIterable inside stat, to avoid creating a second copy of the CAR in mem as `bytes`, until after we scan it to figure out if it is large or invalid or both.